test_regress/t/t_tri_array.out

```diff
@@ -7,19 +7,4 @@
                                          : ... note: In instance 't'
    28 |       Pad pad0 (.pad(pad[g]),
       |                         ^
-%Error-UNSUPPORTED: t/t_tri_array.v:19:17: Unsupported: Inout/tristate with non-bit/logic data type: UNPACKARRAYDTYPE
-                                         : ... note: In instance 't'
-   19 |    tri          pad [NPAD-1:0];   
-      |                 ^~~
-%Error: t/t_tri_array.v:25:25: Select from non-array BASICDTYPE 'bit'
-                             : ... note: In instance 't'
-   25 |       Pad pad1 (.pad(pad[g]),
-      |                         ^
```

Comment:
Try changing test to select from an e.g. queue, or dynamic array?   If there's something that the code can do to trigger this error then it needs to remain a v3error instead of UASSERT in the code.

Reply:
I will try, but I think that the error was thrown due to wrong type (which I fixed in this PR).

Reply:
If I change the type to a queue or dynamic array, an `AstArraySel` node is replaced with `AstCMethodHard` with name `at`. I am not sure, but I think that the `fromp` field of `AstArraySel` should always be of type `AstNodeArrayDType`.

Reply:
Ok, fine to keep the UASSERT then, I wasn't sure.